test/unit/test_recipe.cpp

```diff
@@ -201,20 +201,21 @@ TEST(recipe, event_generators) {
 
     auto cell_0 = custom_cell(1, 2, 0);
     auto cell_1 = custom_cell(2, 1, 0);
-    std::vector<arb::event_generator> gens_0, gens_1;
     {
-        gens_0 = {arb::explicit_generator({{{"synapse0"}, 1.0, 0.1}, {{"synapse1"}, 2.0, 0.1}})};
-
-        gens_1 = {arb::explicit_generator({{{"synapse0"}, 1.0, 0.1}})};
+        std::vector<arb::event_generator>
+            gens_0 = {arb::explicit_generator({"synapse0"}, 0.1, std::vector<arb::time_type>{1.0}),
+                      arb::explicit_generator({"synapse1"}, 0.1, std::vector<arb::time_type>{2.0})},
+            gens_1 = {arb::explicit_generator({"synapse0"}, 0.1, std::vector<arb::time_type>{1.0})};
 
         auto recipe_0 = custom_recipe({cell_0, cell_1}, {{}, {}}, {{}, {}},  {gens_0, gens_1});
         auto decomp_0 = partition_load_balance(recipe_0, context);
 
-        EXPECT_NO_THROW(simulation(recipe_0, context, decomp_0));
+        EXPECT_NO_THROW(simulation(recipe_0, context, decomp_0).run(1, 0.1));
```

Comment:
Why does it need to run now?

Reply:
Actually it always had to. See, the old `event_generator` did up-front target checking and threw an error if any
of them (there could be multiple in one, remember) did not exist. The schedule based one **did not** and did 
defer that to runtime **per event**. 

So, an `event_generator` would have been fine if construction passed, but `schedule_generator` only if simulation
runs. This PR makes the check on schedule eager, removes the need to store the resolver and the runtime checking.
Thus, it's simpler, faster, and safer now.

TL;DR: This requirement for `run` is just paranoia. ;)